src/main/java/gift/service/KakaoService.java

```diff
@@ -1,41 +1,107 @@
 package gift.service;
 
-import java.util.Map;
+import gift.config.JwtTokenProvider;
+import gift.config.KakaoAuthException;
+import gift.domain.AccountType;
+import gift.domain.KakaoToken;
+import gift.domain.Member;
+import gift.dto.KakaoTokenResponse;
+import gift.repository.KakaoTokenJpaRepository;
+import gift.repository.MemberJpaRepository;
+import gift.util.AesUtil;
+import java.net.URI;
+import java.time.LocalDateTime;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpStatus;
 import org.springframework.http.MediaType;
 import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 import org.springframework.util.LinkedMultiValueMap;
 import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.HttpServerErrorException;
 import org.springframework.web.client.RestClient;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.util.UriComponentsBuilder;
 
 @Service
 public class KakaoService {
 
-    @Value("${kakao.api.key}")
-    private String apiKey;
-
-    @Value("${redirect.url}")
-    private String redirectUrl;
+    private final String apiKey;
+    private final String redirectUrl;
+    private final RestClient client;
+    private final AesUtil aesUtil;
+    private final MemberJpaRepository memberRepository;
+    private final JwtTokenProvider jwtTokenProvider;
+    private final KakaoTokenJpaRepository kakaoTokenRepository;
 
+    public KakaoService(
+            @Value("${kakao.api.key}") String apiKey,
+            @Value("${redirect.url}") String redirectUrl,
+            RestClient kakaoRestClient, AesUtil aesUtil, MemberJpaRepository memberRepository,
+            JwtTokenProvider jwtTokenProvider, KakaoTokenJpaRepository kakaoTokenRepository
+    ) {
```

Comment:
```suggestion
    public KakaoService(
            @Value("${kakao.api.key}") String apiKey,
            @Value("${redirect.url}") String redirectUrl,
            RestClient kakaoRestClient, 
            AesUtil aesUtil, 
            MemberJpaRepository memberRepository,
            JwtTokenProvider jwtTokenProvider, 
            KakaoTokenJpaRepository kakaoTokenRepository
    ) {
```
줄바꿈으로 정의할 때는 라인당 하나의 변수를 선언하는게 더 가독성이 좋습니다!